### PR TITLE
stop sending dependencies to /complete

### PR DESCRIPTION
**Pull request by jbergler**
A number of releases ago we started sending this info to /results as well as /complete
At this stage we can remove this from /complete to get one step closer to cleaning this endpoint up.

This is not user facing.

Test plan:
- Run scan locally with supply chain enabled and ensure the backend processes the dependencies (working)

### CODE REVIEW

File: cli/src/semgrep/app/scans.py [hunk 1]
```diff
@@ -358,9 +358,6 @@ def report_findings(
             ),
         )
 
-        if self._dependency_query:
-            complete.dependencies = out.CiScanDependencies(lockfile_dependencies)
```
Comment (aryx) [inline cli/src/semgrep/app/scans.py:380]: can we update semgrep_output_v1.atd and just remove the field there too then?
(removing an optional field is safe; ATD ignores fields he does not know about).

Comment (jbergler) [reply cli/src/semgrep/app/scans.py:380]: Yeah, totally.

Comment (jbergler) [reply cli/src/semgrep/app/scans.py:380]: Well, actually, I think we still support versions that didn't send it to /results and the backend needs to be able to parse it from /complete for now.

Comment (jbergler) [reply cli/src/semgrep/app/scans.py:380]: https://github.com/returntocorp/semgrep-interfaces/pull/148

**Review by aryx — APPROVED** [review]

**Pull request merged**